app/components-react/editor/elements/SourceSelector.tsx

```diff
@@ -632,6 +648,73 @@ class SourceSelectorController {
     return this.scene.getSelection(sceneNodeId);
   }
 
+  toggleDualOutput() {
+    if (this.userService.isLoggedIn) {
+      if (Services.StreamingService.views.isMidStreamMode) {
+        message.error({
+          content: $t('Cannot toggle Dual Output while live.'),
+          className: styles.toggleError,
+        });
+      } else if (Services.TransitionsService.views.studioMode) {
+        message.error({
+          content: $t('Cannot toggle Dual Output while in Studio Mode.'),
+          className: styles.toggleError,
+        });
+      } else {
+        const dualOutputToggleHandled = this.dualOutputService.dualOutputToggleHandled.subscribe(
+          (enabled: boolean) => {
+            console.log('enabled', enabled);
+            if (enabled && this.selectiveRecordingEnabled) {
+              // show warning message if selective recording is active
+              remote.dialog.showMessageBox({
+                title: 'Vertical Display Disabled',
+                message: $t(
+                  'Dual Output can’t be displayed - Selective Recording only works with horizontal sources and disables editing the vertical output scene. Please disable selective recording from Sources to set up Dual Output.',
```

Comment:
Not sure about this message, it looks like it says we couldn't enable dual output but this is only on a subscription, we still proceed to enable below? Most definitely just me not understanding the logic, but wanted to make sure.